packages/services-soap/src/mode/seed.ts

```diff
@@ -39,8 +40,15 @@ export class SoapClientSupplierSeed extends SoapClientSupplierBase {
       const wsdl = await this.fetchWsdl(opts)
       const endpoint = getEndpoint(opts)
       if (!wsdlIsFile(opts)) {
-        const seed = FxSt.createByWsdlUrl(endpoint)
-        seed.set(wsdl)
+        const seed = FxSt.createByWsdlUrl<string>(endpoint)
+        try {
+          seed.get()
+          if (seed.expired) {
+            throw new Error()
+          }
+        } catch {
+          seed.set(wsdl)
+        }
```

Comment:
is it necessary to throw an exception if expired? In my feeling exceptions handle situation that cannot be foreseen (like a third party system is down, etc.). Expiration is a standard case and should be embedded in the default flow.

Reply:
performance-wise it should be equal, the difference would be the control flow:
````typescript
const setSeed = (opts) => {
  const wsdlString = await this.getWsdl(opts)
  seed.set(wsdlString)
}
try {
  seed.get()
  if (seed.expired) {
    setSeed(opts)
  }
} catch {
  setSeed(opts)
}
````
i think the question should be if the expired-logic wouldn't be better put into the fixture storage logic, as this is practically copied in services-http and -soap.